ts/src/test/test.ts

```diff
@@ -289,34 +275,40 @@ export default class testMainClass extends baseMainTestClass {
                 await this.testMethod (methodName, exchange, args, isPublic);
                 return true;
             } catch (e) {
+                const isAuthError = (e instanceof AuthenticationError);
                 const isRateLimitExceeded = (e instanceof RateLimitExceeded);
                 const isExchangeNotAvailable = (e instanceof ExchangeNotAvailable);
                 const isNetworkError = (e instanceof NetworkError);
                 const isDDoSProtection = (e instanceof DDoSProtection);
                 const isRequestTimeout = (e instanceof RequestTimeout);
                 const tempFailure = (isRateLimitExceeded || isExchangeNotAvailable || isNetworkError || isDDoSProtection || isRequestTimeout);
                 if (tempFailure) {
-                    // wait and retry again
-                    await exchange.sleep (i * 1000); // increase wait seconds on every retry
                     // if last retry was gone with same `tempFailure` error, then let's eventually return false
                     if (i === maxRetries - 1) {
                         dump ('[TEST_WARNING]', 'Method could not be tested due to a repeated Network/Availability issues', ' | ', exchange.id, methodName, argsStringified);
-                        if (methodName === 'loadMarkets') {
-                            // in case of loadMarkets, we completely stop test for current exchange
-                            exitScript ();
-                        }
-                        return false;
+                    } else {
+                        // wait and retry again
+                        await exchange.sleep (i * 1000); // increase wait seconds on every retry
+                        continue;
                     }
-                    continue;
                 } else if (e instanceof OnMaintenance) {
                     // in case of maintenance, skip exchange (don't fail the test)
                     dump ('[TEST_WARNING] Exchange is on maintenance', exchange.id);
-                    exitScript ();
+                }
+                // If public test faces authentication error, we don't break (see comments under `testSafe` method)
+                else if (isPublic && isAuthError) {
+                    if (this.info) {
+                        dump ('[TEST_WARNING]', 'Authentication problem for public method', exceptionMessage (e), exchange.id, methodName, argsStringified);
+                    }
                 } else {
                     // if not a temporary connectivity issue, then mark test as failed (no need to re-try)
                     dump ('[TEST_FAILURE]', exceptionMessage (e), exchange.id, methodName, argsStringified);
-                    return false;
                 }
+                if (methodName === 'loadMarkets') {
+                    // we throw exception only for `loadMarkets` failed test
+                    throw e;
```

Comment:
in case of `loadMarkets` failed test, we completely stop tests for exchange with exception (no point to continue testing of other methods)